### PR TITLE
Add redux

### PR DESCRIPTION
**Pull request by wale-prog**
In this pull request, I have implemented the following:
- add Redus Toolkit and structure my application files using the `ducks pattern`
- write the book's actions, then export action creators and reducer to be used in other component files
- defined action types for adding and removing books.
- wrote categories action and reducer then exported them
- Redux store has been created and necessary methods imported from Redux Toolkit.
- the reducers for both books and categories have been combined using the `configureStore` function.

All these and more have been implemented

Thanks.

### CODE REVIEW

File: src/redux/books/books.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
+/* eslint-disable */
+import { v4 as uuidv4 } from 'uuid';
+const ADD_BOOK = "ADD_BOOK";
```
Comment (topeogunleye) [inline src/redux/books/books.js]: - [ ] I kindly suggest that you follow the [ducks](https://github.com/erikras/ducks-modular-redux) pattern 

Wrong :x: 
`const ADD_BOOK = "ADD_BOOK";
`

Right :heavy_check_mark: 
`const ADD_BOOK = bookstore/books/ADD_BOOK`

File: src/redux/categories/categories.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
+const CHECK_STATUS = 'CHECK_STATUS';
```
Comment (topeogunleye) [inline src/redux/categories/categories.js]: - [ ] I kindly suggest that you follow the [ducks](https://github.com/erikras/ducks-modular-redux) pattern 

Wrong :x: 
`const CHECK_STATUS = 'CHECK_STATUS';`

Right :heavy_check_mark: 
`const CHECK_STATUS = bookstore/categories/CHECK_STATUS`

**Review by topeogunleye — CHANGES_REQUESTED** [review]
> # Changes Required 
> ### Hi @wale-prog,
> Nice job, you are almost there.
> 
> 
> 
> ## (Highlights) Good Points: 
> * Good commit messages. :heavy_check_mark:
> * No linter errors. :heavy_check_mark:
> * Followed Github flow :heavy_check_mark:
> * Professional README file. :heavy_check_mark:
> * PR has a good title and summary :heavy_check_mark:
>
> ## (Changes Required) Aspects to improve: 
> * [ ]  Please check my inline comments.
> 
> ## [Optional] suggestions:
> * Nothing to mention.
> 
> _Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
> 
> **Cheers and Happy coding! :clap::clap::clap:**
> 
> Feel free to leave any questions or comments in the PR thread if something is not 100% clear
> Please, remember to tag @topeogunleye in your question so I can receive the notification.
>
> **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
> 
> As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by ShadyShawkat — APPROVED** [review]
## Hi @wale-prog, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Image

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**